hbase-procedure/src/main/java/org/apache/hadoop/hbase/procedure2/ProcedureExecutor.java

```diff
@@ -1441,42 +1466,73 @@ private void releaseLock(Procedure<TEnvironment> proc, boolean force) {
     }
   }
 
-  /**
-   * Execute the rollback of the full procedure stack. Once the procedure is rolledback, the
-   * root-procedure will be visible as finished to user, and the result will be the fatal exception.
-   */
-  private LockState executeRollback(long rootProcId, RootProcedureState<TEnvironment> procStack) {
-    Procedure<TEnvironment> rootProc = procedures.get(rootProcId);
-    RemoteProcedureException exception = rootProc.getException();
-    // TODO: This needs doc. The root proc doesn't have an exception. Maybe we are
-    // rolling back because the subprocedure does. Clarify.
-    if (exception == null) {
-      exception = procStack.getException();
-      rootProc.setFailure(exception);
-      store.update(rootProc);
+  private IdLock.Entry getLockEntryForRollback(long procId) {
```

Comment:
Do we need docs here? Explain the meaning of null.

Reply:
Done